subsys/cpp/cpp_vtable.cpp

```diff
@@ -14,6 +14,7 @@
  * @brief basic virtual tables required for classes to build
  *
  */
+#if !defined(CONFIG_LIB_CPLUSPLUS)
```

Comment:
`__cxxabiv1` is the namespace used by GNU's libstdc++, fine.
A comment about this would be good.